### ISSUE TITLE
Close note on Middle click, Create new on Double click

### ISSUE THREAD

**Issue by macsunmood** [pbek/QOwnNotes#2569]
#### Expected behaviour
Add some UI improvements to Tabs panel: 
- Close note under mouse cursor with Middle click over its Tab;
- Quickly create new note with Double click anywhere over Tabs panel free space (as it's implemented in Akelpad).

**Comment by pbek** [comment]
Thank you for your suggestions.

**Comment by starze** [comment]
I would really love to see this feature added to QOwnNotes. While these kinds of small tweaks may not seem like major features, they are usually easy to implement and, in my opinion, they make a big difference in the overall user experience.

Perhaps this snippet can serve as a useful starting point for one of the main developers (unfortunately, I don’t currently have a working development environment to test it myself).

```cpp
void MainWindow::setupMiddleClickClose()
{
    QTabBar *tabBar = ui->noteEditTabWidget->tabBar();
    tabBar->installEventFilter(this); // Install an event filter on the tab bar
}

bool MainWindow::eventFilter(QObject *obj, QEvent *event)
{
    if (obj == ui->noteEditTabWidget->tabBar() && event->type() == QEvent::MouseButtonPress) {
        QMouseEvent *mouseEvent = static_cast<QMouseEvent *>(event);
        if (mouseEvent->button() == Qt::MiddleButton) {
            int tabIndex = ui->noteEditTabWidget->tabBar()->tabAt(mouseEvent->pos());
            if (tabIndex != -1) {
                ui->noteEditTabWidget->removeTab(tabIndex); // Close the tab
                return true; // Event handled
            } else {
                // create new tab
                // ...
            }
        }
    }
    return QMainWindow::eventFilter(obj, event);
}
```

**Comment by pbek** [comment]
@starze, thank you for the suggestion!

> Quickly create new note with Double click anywhere over Tabs panel free space (as it's implemented in Akelpad).

@macsunmood, what "over Tabs panel free space" are we talking about?
The space next to the open tabs?

**Comment by pbek** [comment]
## 25.5.0

- If you middle-click on a note tab, it will now close (for [#2569](https://github.com/pbek/QOwnNotes/issues/2569))
- If you double-click on the empty area next to the note tabs, a new note in a
  new tab it will be opened (for [#2569](https://github.com/pbek/QOwnNotes/issues/2569))

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if it works for you?

**Comment by macsunmood** [comment]
> [@starze](https://github.com/starze), thank you for the suggestion!
> 
> > Quickly create new note with Double click anywhere over Tabs panel free space (as it's implemented in Akelpad).
> 
> [@macsunmood](https://github.com/macsunmood), what "over Tabs panel free space" are we talking about? The space next to the open tabs?

Hi! Yes, you got that exactly right! 
Thank you for implementing these features! 👍🔥

**Comment by pbek** [comment]
Great, thank you for testing!

**Comment by starze** [comment]
> There now is a new release, could you please test it and report if it works for you?
@pbek, Thanks a lot! As @macsunmood said, you got that exactly right!

My initial tests looked promising, but when I open more tabs (around 5–10) and middle-click them quickly, I sometimes experience unexpected behavior. Instead of closing the tab, it occasionally flags the clicked tab. It seems like rapid middle-clicking is being interpreted as a double-click, which triggers the flagging instead of the intended close action.

My first guess is that the `return true;` statement should be placed outside the if-statements, as the [on_noteEditTabWidget_tabBarDoubleClicked](https://github.com/pbek/QOwnNotes/commit/c1380abc1dba591a8a7c0d7bb237e88227011256#diff-c72b037f3ac7abbd3d0ee519b6f456013a9bbf5a59df16308939320c04970447L11984) function seems to be triggered independently.

Please let me know if I should open a new issue for this instead

**Comment by pbek** [comment]
Yes, `on_noteEditTabWidget_tabBarDoubleClicked` catches every double-click, regardless what key, and there is no way of determining which key was pressed. And it works independently of `eventFilter´, so returning a `true` in `eventFilter´ doesn't stop `on_noteEditTabWidget_tabBarDoubleClicked` being fired (I just tested).

**Comment by starze** [comment]
Oh, that’s a pity, but I can live with it for now. Thanks again for implementing the feature - I really appreciate it.

**Comment by pbek** [comment]
Thank you and thank you for reporting anyway! 😁
